[86bztnr9z][carousel] navigatable carusel controls

msereniti opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Motivation and Context

Basically, this PR reduced carousel usability from screen reader but now it will comply with formal requirments of a11y audits.

## How has this been tested?

Manually.

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Nice improve.

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly or it's not required.
- [x] Unit tests are not broken.
- [x] I have added changelog note to corresponding `CHANGELOG.md` file with planned publish date.
- [ ] I have added new unit tests on added of fixed functionality.

ilyabrower commented on semcore/carousel/CHANGELOG.md:
```diff
@@ -2,6 +2,12 @@
+
+### Changed
+
+- Carousel controls are not focusable and available to be navigated to each control by screen readers.
```
maybe on present - `Carousel controls are focusable and available...` ?
or change `Changed` to `Fixed`?

j-mnizhek commented:
> Have a couple of comments:
> 
> 1. `Carousel.Indicators` with `tablist` role lack `aria-label="Slides"`.
> 2. When we set `zoom` property for the Carousel to false, images are still have the button role and read with the aria-label from the Button.
> 3. Btw, why Carousel.Container has the `aria-roledescription="carousel"`? Isn't the Carousel having this `aria-roledescription="carousel"` attribute not enought? I also didn't see the requirement to have such an attribute on the Carousel.Container.

@msereniti Misha, can you please check these first and explain why something cannot be done? Then I will be able to update or not update oir documentation 🙏

j-mnizhek reviewed (changes requested):
Left comment with a couple of questions.

msereniti commented:
>Carousel.Indicators with tablist role lack aria-label="Slides".

Sent aria labels to translations and added it usages to code

> When we set zoom property for the Carousel to false, images are still have the button role and read with the aria-label from the Button.

This is on the side of example, specific developers will control it

> Btw, why Carousel.Container has the aria-roledescription="carousel"? Isn't the Carousel having this aria-roledescription="carousel" attribute not enought? I also didn't see the requirement to have such an attribute on the Carousel.Container.

Mistake, fixed it

j-mnizhek commented:
Noticed that I cannot open in fullscreen using `Enter` the `Carousel.Item` when it's focused.

j-mnizhek reviewed (approved):
Updated docs a little bit.

sheila-semrush commented on website/docs/components/carousel/carousel-a11y.md:
```diff
@@ -22,25 +22,30 @@ The list below describes roles and attributes that component already has.
+| --------------------- | --------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `Carousel`            | `role="region"`, `roledescription="carousel"` | Defines the carousel and its controls as a landmark region. `roledescription` informs assistive technologies to identify the element as a "carousel".                                |
+| `Carousel.Indicators` | `role="tablist"`                              | Indicates that the element serves as a container for a set of tabs.                                                                                                                  |
+|                       | `aria-label="Slides"`                         | Adds a label to a container for a set of tabs.                                                                                                                                       |
```
@j-mnizhek I've been thinking about this recently, wouldn't the empty cells be an a11y issue because it's not clear that we mean a merged cell here?

j-mnizhek replied:
This is an interesting question! Let me consult with Glasha first, maybe she have had something similar in her experience

sheila-semrush commented on website/docs/components/carousel/carousel-a11y.md:
```diff
@@ -13,7 +13,7 @@ Table: Keyboard support
 | --------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `Tab`, `Shift + Tab`        | Moves focus through interactive elements in the carousel.                                                                                                                          |
-| `Enter`, `Space`            | Opens slide in modal window if `zoom` property is set to `true`.                                                                                                                   |
+| `Enter`                     | Opens slide in modal window if `zoom` property is set to `true`.                                                                                                                   |
```
hmmm it might be an issue. It has role=button and buttons are supposed to activate on Space

j-mnizhek replied:
Yep, it was some strange behaviour, I've wrote Michael about. If it will finally works without Space scrolling the page, I'll leave the `Space`.

sheila-semrush commented on website/docs/components/carousel/carousel-a11y.md:
```diff
@@ -13,7 +13,7 @@ Table: Keyboard support
 | `Tab`, `Shift + Tab`        | Moves focus through interactive elements in the carousel.                                                                                                                          |
-| `Enter`, `Space`            | Opens slide in modal window if `zoom` property is set to `true`.                                                                                                                   |
+| `Enter`                     | Opens slide in modal window if `zoom` property is set to `true`.                                                                                                                   |
 | `Left Arrow`, `Right Arrow` | Moves focus to the previous or next tab in the slide indicators. If focus is on the last tab, moves focus to the first tab. Shows the slide associated with the newly focused tab. |
```
```suggestion
| `Left Arrow`, `Right Arrow` | Activates the previous or next tab in the slide indicators. If the last tab is active, activates the first tab. Shows the slide associated with the activated tab. |
```
I think it's better to avoid the word "focus" here, because as I see the focus is always on the entire indicator row. And maybe ever remove the last sentence, it might be redundant.

j-mnizhek replied:
Yeah! You are definitely right here, thank you ✨

sheila-semrush commented on website/docs/components/carousel/carousel-a11y.md:
```diff
@@ -22,25 +22,30 @@ The list below describes roles and attributes that component already has.
+| `Carousel`            | `role="region"`, `roledescription="carousel"` | Defines the carousel and its controls as a landmark region. `roledescription` informs assistive technologies to identify the element as a "carousel".                                |
+| `Carousel.Indicators` | `role="tablist"`                              | Indicates that the element serves as a container for a set of tabs.                                                                                                                  |
+|                       | `aria-label="Slides"`                         | Adds a label to a container for a set of tabs.                                                                                                                                       |
+| `Carousel.Indicator`  | `role="tab"`                                  | Indicates the element serves as a tab control. When focused, it is automatically activated, causing its associated tabpanel to be displayed.                                         |
```
```suggestion
| `Carousel.Indicator`  | `role="tab"`                                  | Indicates the element serves as a tab control. When active, the associated tabpanel is automatically displayed.                                         |
```
same motivation as above

sheila-semrush commented on website/docs/components/carousel/carousel-a11y.md:
```diff
@@ -22,25 +22,30 @@ The list below describes roles and attributes that component already has.
+| `Carousel.Item`       | `role="tabpanel"`                             | Indicates that the element acts as a tabpanel, controlled by its linked tab.                                                                                                         |
+|                       | `aria-roledescription="slide"`                | Indicates the element serves as a tabpanel that is controlled by its associated tab element.                                                                                         |
+|                       | `aria-current="active"`                       | Indicates that this element represents the current item within a container or set of related elements.                                                                               |
+| `Carousel.Prev`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the tab.                                                                                                                              |
```
```suggestion
| `Carousel.Prev`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the previous tab.                                                                                                                              |
```
might be worth clarifying

sheila-semrush commented on website/docs/components/carousel/carousel-a11y.md:
```diff
@@ -22,25 +22,30 @@ The list below describes roles and attributes that component already has.
+|                       | `aria-current="active"`                       | Indicates that this element represents the current item within a container or set of related elements.                                                                               |
+| `Carousel.Prev`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the tab.                                                                                                                              |
+|                       | `aria-label="Previous slide"`                 | Defines a label for the button that opens previous list item.                                                                                                                        |
+| `Carousel.Next`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the tab.                                                                                                                              |
```
```suggestion
| `Carousel.Next`       | `aria-controls="carouselID"`                  | Refers to the tabpanel element associated with the next tab.                                                                                                                              |
```
same as with Prev
